platforms/documentation/docs/src/docs/userguide/dep-man/declaring-depedencies/declaring_configurations.adoc

```diff
@@ -0,0 +1,148 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[dependency-configurations]]
+= Declaring Dependency Configurations
+
+[[sec:what-are-dependency-configurations]]
+In Gradle, dependencies are associated with specific scopes, such as compile-time or runtime.
+These scopes are represented by configurations, each identified by a unique name.
+
+image::dependency-management-configurations.png[]
+
+Gradle plugins often add pre-defined link:{groovyDslPath}/org.gradle.api.artifacts.Configuration.html[configuration] to your project.
+For example, the Java plugin adds configurations for source code compilation, test execution, and more:
+
+====
+include::sample[dir="snippets/dependencyManagement/customizingResolution-classifierSubstitutionRule/kotlin",files="consumer/build.gradle.kts[tags=dependencies]"]
+include::sample[dir="snippets/dependencyManagement/customizingResolution-classifierSubstitutionRule/groovy",files="consumer/build.gradle[tags=dependencies]"]
+====
+
+This example highlights project and module dependencies declared on the `implementation` configuration.
+In a Java project, the `implementation` configuration is used to specify dependencies required to compile and run the main source code.
+See the <<java_plugin.adoc#sec:java_plugin_and_dependency_management,Java plugin documentation>> for details.
+
+[[sub:config-inheritance-composition]]
+== Understanding configuration inheritance
+
+Configurations can inherit from other configurations, creating an inheritance hierarchy.
+Child configurations inherit all dependencies declared in their parent configurations.
+
+Configuration inheritance is commonly used by Gradle core plugins, such as the <<java_plugin.adoc#sec:java_plugin_and_dependency_management,Java plugin>>.
+For instance, the `testImplementation` configuration extends the `implementation` configuration.
+
+image::dependency-management-configuration-inheritance.png[]
+
+This hierarchy ensures that dependencies needed for both compiling tests and the source code under test are included.
+For example, a Java project using JUnit for tests might also need Guava if it's used in the production code.
+
+Internally, configurations form an inheritance hierarchy using the link:{groovyDslPath}/org.gradle.api.artifacts.Configuration.html#org.gradle.api.artifacts.Configuration:extendsFrom(org.gradle.api.artifacts.Configuration++[]++)[`Configuration.extendsFrom(Configuration...)`] method.
+A configuration can extend any other configuration, regardless of how it is defined in the build script or plugin.
+
+For instance, if you want to create a suite of smoke tests that verify a web service endpoint using JUnit, you can define a new configuration named `smokeTest` that extends from `testImplementation` to reuse the existing test framework dependency:
```

Comment:
With test suites we explicitly avoided this sort of pattern so I'm not sure its a great way to showcase extendsFrom in the docs

Reply:
TODO: This needs a small discussion on what good examples we have.

Reply:
Complete overhaul of the page complete following discussion.